src/main/rst/examples/Chapter4/Java/ProxySetupFirefoxMarionette.java

```diff
@@ -0,0 +1,15 @@
+String PROXY = "localhost";
+int PORT = 8080;
+
+com.google.gson.JsonObject json = new com.google.gson.JsonObject();
+json.addProperty("proxyType", "MANUAL");
+json.addProperty("httpProxy", PROXY);
+json.addProperty("httpProxyPort", PORT);
+json.addProperty("sslProxy", PROXY);
+json.addProperty("sslProxyPort", PORT);
+
+DesiredCapabilities desired = DesiredCapabilities.firefox();
+DesiredCapabilities required = new DesiredCapabilities();
+required.setCapability("proxy", json);
+
+driver = new FirefoxDriver(desired, required);
```

Comment:
the only difference between this and the other is to pass in required capabilities... 

So, why don't we just change that example to pass in the 'cap' to both desired and required and just put a comment like this:

    // GeckoDriver currently needs the Proxy set in RequiredCapabilities, this will work for both 
    // GeckoDriver and Selenium's Firefox extension webdriver.xpi
    WebDriver driver = new FirefoxDriver(cap, cap);